scalafix-rules/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -173,7 +173,12 @@ final class ExplicitResultTypes(
     def hasParentWihTemplate: Boolean =
       defn.parent.exists(_.is[Template])
 
-    isImplicit && !isFinalLiteralVal || {
+    {
+      isImplicit &&
+      !isFinalLiteralVal &&
+      matchesMemberKind() &&
+      matchesMemberVisibility()
```

Comment:
these two functions could be extracted out instead of being repeated in both the blocks which are in OR